docs/OV_Runtime_UG/stateful_models_intro.md

```diff
@@ -0,0 +1,90 @@
+Stateful models and State API {#openvino_docs_OV_UG_stateful_models_intro}
+==============================
+
+@sphinxdirective
+
+.. toctree::
+    :maxdepth: 1
+    :hidden:
+
+    openvino_docs_OV_UG_ways_to_get_stateful_model
+
+@endsphinxdirective
+
+## What is a Stateful Network?
+
+ Several use cases require processing of data sequences. When length of a sequence is known and small enough, 
+ we can process it with RNN like networks that contain a cycle inside. But in some cases, like online speech recognition of time series 
+ forecasting, length of data sequence is unknown. Then data can be divided in small portions and processed step-by-step. But dependency 
+ between data portions should be addressed. For that, networks save some data between inferences - state. When one dependent sequence is over,
+ state should be reset to initial value and new sequence can be started.
+ 
+ Several frameworks have special API for states in networks. For example, Keras has special option for RNNs `stateful` that turns on saving state 
+ between inferences. Kaldi contains special specifier `Offset` to define time offset in a network. 
+ 
+ OpenVINO also contains special API to simplify work with networks with states. State is automatically saved between inferences, 
+ and there is a way to reset state when needed. You can also read state or set it to some new value between inferences.
+ 
+## OpenVINO State Representation
+
+ OpenVINO contains a special abstraction `Variable` to represent a state in a network. There are two operations to work with the state: 
+* `Assign` to save value in state
+* `ReadValue` to read value saved on previous iteration
+
+![state_network_example](./img/state_network_example.png)
+
+You can find more details on these operations in [ReadValue specification](../ops/infrastructure/ReadValue_3.md) and 
+[Assign specification](../ops/infrastructure/Assign_3.md).
+
```

Comment:
I recommend to move this part below after the API description. User is rarely involved in model representation internals. Let's provide the most relevant information first then all such details.